packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -508,6 +520,24 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
     return FlutterCommandResult.success();
   }
 
+  Iterable<String> _getCoveragePackages(
+    List<String> packagesRegExps,
+    FlutterProject flutterProject,
+    PackageConfig packageConfig,
+  ) {
+    final String projectName = flutterProject.manifest.appName;
+    final Iterable<String> packagesToInclude;
+    if (packagesRegExps.isNotEmpty) {
+      final RegExp combinedRegExp = RegExp('(${packagesRegExps.join(")|(")})');
```

Comment:
I'd prefer if you do one test per regex and then join the results at the end, rather than combining it into one large regular expression.